framework/auth/core.py

```diff
@@ -855,6 +860,23 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def confirm_token(self, token):
+        """Return whether or not a confirmation token is valid for this user.
+        :rtype: bool
+        """
+        try:
+            verification = self.email_verifications[token]
+        except KeyError:
+            return False
+        # Check token for existance and date
+        if (
+            'expiration' in verification and
+            (verification['expiration'] < dt.datetime.utcnow() or
+                token not in self.email_verifications)
```

Comment:
I think `or token not in self.email_verifications` would be a noop here, as you've already bailed out if `self.email_verifications[token]` raises a `KeyError`.